content/automate/ManageIQ/Transformation/TransformationHosts/Common.class/__methods__/utils.rb

```diff
@@ -31,20 +33,85 @@ def self.transformation_hosts(ems, method, factory_config)
               thosts.sort_by! { |th| th[:runners][:current] }
             end
 
-            def self.eligible_transformation_hosts(ems, method, factory_config)
-              transformation_hosts(ems, method, factory_config).select { |thost| thost[:runners][:current] < thost[:runners][:maximum] }
+            def self.eligible_transformation_hosts(ems, factory_config)
+              transformation_hosts(ems, factory_config).select { |thost| thost[:runners][:current] < thost[:runners][:maximum] }
             end
 
-            def self.get_runners_count_by_ems(ems, method, factory_config)
-              transformation_hosts(ems, method, factory_config).inject(0) { |sum, thost| sum + thost[:runners][:current] }
+            def self.get_runners_count_by_ems(ems, factory_config)
+              transformation_hosts(ems, factory_config).inject(0) { |sum, thost| sum + thost[:runners][:current] }
             end
 
-            def self.get_transformation_host(ems, method, factory_config)
-              ems_max_runners = ems.custom_get('Max Transformation Runners') || factory_config['ems_max_runners'] || 1
-              ems_cur_runners = get_runners_count_by_ems(ems, method, factory_config)
-              transformation_host = ems_cur_runners < ems_max_runners ? eligible_transformation_hosts(ems, method, factory_config).first[:host] : nil
-              transformation_host
+            def self.get_transformation_host(task, factory_config, handle = $evm)
```

Comment:
I would prefer to revert `task` to `ems`. As a utility method it is better to accept more direct objects. Same for other methods in this class where `task` is accepted.

Reply:
ok